media-source-respec.html

```diff
@@ -757,38 +766,70 @@ <h4><dfn>Attaching to a media element</dfn></h4>
+          in the "<i>Otherwise</i>" clause, below, until the MediaSource object is ready for use.</p>
           <dl class="switch">
-            <dt>If the media provider object is
-              a URL record whose object is a {{MediaSource}} that was constructed in a {{DedicatedWorkerGlobalScope}},
-              such as would occur if attempting to use  a [=MediaSource object URL=] from a
-              {{DedicatedWorkerGlobalScope}} {{MediaSource}}
+            <dt>If the media provider object is a URL record whose object is a {{MediaSource}} that was constructed in a
+              {{DedicatedWorkerGlobalScope}}, such as would occur if attempting to use a [=MediaSource object URL=]
+              from a {{DedicatedWorkerGlobalScope}} {{MediaSource}}
               </dt>
-            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the <a def-id="resource-fetch-algorithm"></a>'s <a def-id="media-data-processing-steps-list"></a>.</dd>
-            <dt>If {{MediaSource/readyState}} is NOT set to {{ReadyState/""closed""}} 
-            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the <a def-id="resource-fetch-algorithm"></a>'s <a def-id="media-data-processing-steps-list"></a>.</dd>
+            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the [=resource fetch algorithm=]'s
+              <a def-id="media-data-processing-steps-list"></a>.
+              <div class="note">This prevents using [=MediaSource object URLs=] for DedicatedWorker MediaSource
+                attachments. Transferring {{MediaSource}}.{{MediaSource/handle}} from the DedicatedWorker to the Window
+                context and assigning it to the media element's {{HTMLMediaElement/srcObject}} attribute is the only way
+                to attach such a MediaSource.
+              </div>
+            </dd>
+            <dt>If the media provider object is a {{MediaSourceHandle}} whose underlying {{MediaSource}} has ever
+              been used beyond this point by in any {{HTMLMediaElement}}'s [=resource fetch algorithm=]</dt>
+            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the [=resource fetch algorithm=]'s
+              <a def-id="media-data-processing-steps-list"></a>.
+              <div class="note">This prevents loading an underlying {{MediaSource}} more than once using a
+                {{MediaSourceHandle}}, even if the {{MediaSource}} was constructed on {{Window}} and had been loaded
+                previously using a [=MediaSource object URL=]. This doesn't preclude subsequent use of a [=MediaSource
+                object URL=] for a {{Window}} {{MediaSource}} from succeeding though.
+              </div>
+            </dd>
+            <dt>If the media provider object is a {{MediaSource}} that is already being used
+              beyond this point as the media provider object in a different {{HTMLMediaElement}}'s [=resource fetch
```

Comment:
"being used beyond this point" is a little more ambiguous than "been used beyond this point" and again I think it would be better if there was some explicit state that could be consulted.

Reply:
"If {{MediaSource/readyState}} is NOT set to {{ReadyState/""closed""}}" below is catching the same case as "{{MediaSource}} that is already being used beyond this point" right?
I prefer the more specific `readyState` check.

Reply:
We also need to block a `MediaSourceHandle` with [[Detached]] true, but that need not be part of this PR.

Reply:
> "being used beyond this point" is a little more ambiguous than "been used beyond this point" and again I think it would be better if there was some explicit state that could be consulted.

I agree. Also, @karlt commented similarly. The state for this clause already exists: readyState. I'll remove this redundant clause since there already is a readyState check that pre-existed the MSE-in-Workers changes.

Reply:
> "If {{MediaSource/readyState}} is NOT set to {{ReadyState/""closed""}}" below is catching the same case as "{{MediaSource}} that is already being used beyond this point" right?
> I prefer the more specific `readyState` check.

I agree. I'll remove this redundant "being used beyond this point" clause.

Reply:
> We also need to block a `MediaSourceHandle` with [[Detached]] true, but that need not be part of this PR.

This is a good addition to include. WPT already test for this and Chromium experimental implementation already has this. I'll add it in this PR.

Reply:
Done in 6e927afbbc43d047301031f28cf95b811cdb80d0